### ISSUE TITLE
HOI-A dataset

### ISSUE THREAD

**Issue by SherlockHolmes221** [YueLiao/PPDM#8]
Where I can get the images and annotations of HOI-A dataset?
Thanks.

**Comment by YueLiao** [comment]
We will release the HOI-A in this repo in early April. It may be a little delay due to the covid-19.

**Comment by SherlockHolmes221** [comment]
I found some similarity between the HOI-A and HOI-W dataset, are they the same dataset?

**Comment by YueLiao** [comment]
Our HOI-A dataset is built upon our HOI-W dataset for ICCV HOI detection challenge with some refinements and extensions. There is a little difference, e.g., lots of wrong annotations in HOI-W are fixed in HOI-A, and some negative samples are added.  The performance of our PPDM on HOI-W and HOI-A has a little difference (about 1% point).

**Comment by SherlockHolmes221** [comment]
Thanks, closing it

**Comment by Agile-yun** [comment]
I read your interesting paper, and I try to train/test your model. When can I get the HOI-A Dataset? If it takes a long time, I want to test using the HOI-W dataset. Can I upload it to Google drive other than Baidu Drive?

**Comment by YueLiao** [comment]
OneDrive is OK? I don't have enough space to upload and store the HOI-W dataset on Google Drive. I have finished my deadlines just now, and I will try to release the HOI-A dataset recently.

**Comment by Agile-yun** [comment]
Onedrive is also good. Thanks.

**Comment by YueLiao** [comment]
> Onedrive is also good. Thanks.

https://sensetime-my.sharepoint.cn/:f:/g/personal/liaoyue_sensetime_com/EkUA1sc_sHNNmVNiVsyF8L0BCE4AXHa7sFEb8YTx11laBg?e=ArfMCD

**Comment by deep-practice** [comment]
When will the dataset be released?

**Comment by zhazhazhazhi** [comment]
I  find that the MAP value is the 0.2  in the HOI-W .

**Comment by YueLiao** [comment]
> I  find that the MAP value is the 0.2  in the HOI-W . 

Can you provide more details？ I can achieve 68.1 mAP based on DLA backbone using the repo.

**Comment by YueLiao** [comment]
> When will the dataset be released?

I am sorry for the delay. I planned to released the HOI-A this week，but the covid-2019 outbreaks in Beijing which causes that I cannot return to the office in Beijing. Once I can return to the office，the HOI-A dataset will be in public in a week.

**Comment by zhazhazhazhi** [comment]
1.according to the corre_hico.npy ( modify  suit for hoiw  )  . office label in hoiw . 
[1. 1. 1. 1. 0. 0. 0. 0. 0. 0. 0.]
[1. 1. 1. 1. 1. 1. 1. 0. 0. 1. 1.]
[1. 1. 1. 1. 1. 1. 1. 0. 0. 1. 1.]
[1. 1. 1. 1. 1. 0. 0. 0. 0. 0. 0.]
[1. 1. 1. 1. 1. 0. 0. 0. 0. 0. 0.]
[1. 1. 0. 1. 0. 1. 1. 1. 0. 0. 1.]
[1. 1. 1. 1. 1. 1. 1. 0. 1. 1. 1.]
[1. 0. 0. 0. 0. 0. 0. 0. 1. 0. 0.]
[1. 1. 0. 1. 1. 0. 0. 0. 0. 1. 1.]
[1. 1. 0. 1. 0. 0. 0. 0. 0. 1. 1.]    (corre_hico.npy)
2. then I copy your hoia file and rename  to hoiw file . 
 example :(  
(1) dataset /   dataset_factory.py  add the hoiw . All hoia files are modified into  hoiw files  
(2) dataset / dataset   hoia.py     class HOIA(Dataset)   -  class HOIWDataset)  ,and so on 
(3) opts.py add          self.parser.add_argument('--dataset', default='hoiw',
                                 help='hico | vcoco | hoia| hoiw')
         if opt.task == 'hoidet':
            assert opt.dataset in ['hico', 'vcoco', 'hoia', 'hoiw']
            opt.heads = {'hm': opt.num_classes,
                         'wh': 2 if not opt.cat_spec_wh else 2 * opt.num_classes,
                         'hm_rel': opt.num_classes_verb,
                         'sub_offset': 2,
                         'obj_offset': 2}
(4)  detectors / hoidet.py 
        elif 'hoiw' in opt.dataset:
            self.corre_mat = np.load(os.path.join(self.opt.root_path, 'hoia/annotations/corre_hoiw.npy'))

        if 'hoiw' in self.opt.dataset:
            obj_cate_ids= list(range(1, 12))
            verb_cate_ids = list(range(1, 11))
 )
3. train . (140epoch,DLA backbone. ) 
 python main.py  hoidet --batch_size 50 --master_batch 7 --lr 4.5e-4 --gpus 0,1,2,3  --num_workers 16  --load_model ../models/dla34/model_best.pth( your provided  ) --image_dir trainval/trainval --dataset hoiw --exp_id hoidet_hico_dla 
    but now I train the  model  ( loading model ../models/ctdet_coco_dla_2x.pth ) 

4. test_hoi  python test_hoi.py hoidet --exp_id hoidet_hico_dla --load_model ../models/w-dla34/model_140.pth --gpus 0 --dataset hoiw --image_dir trainval/test --test_with_eva

5.results :mAP: 0.2061473138228045   max recall: 0.3803719110411382 

the enviorment is fine . (python 3.6  , pytorch 0.4.1 )

**Comment by zhazhazhazhi** [comment]
> > I  find that the MAP value is the 0.2  in the HOI-W .
> 
> Can you provide more details？ I can achieve 68.1 mAP based on DLA backbone using the repo.

But I find the error : in the ../models/ctdet_coco_dla_2x.pth   .
 Skip loading parameter hm.2.weight, required shapetorch.Size([11, 256, 1, 1]), loaded shapetorch.Size([80, 256, 1, 1]).
Skip loading parameter hm.2.bias, required shapetorch.Size([11]), loaded shapetorch.Size([80]).
No param hm_rel.0.weight.
No param hm_rel.0.bias.
No param hm_rel.2.weight.
No param hm_rel.2.bias.
No param sub_offset.0.weight.
No param sub_offset.0.bias.
No param sub_offset.2.weight.
No param sub_offset.2.bias.
No param obj_offset.0.weight.
No param obj_offset.0.bias.
No param obj_offset.2.weight.
No param obj_offset.2.bias.

Can you provide your pre-trained model in HOIW or HOIA dataset?

**Comment by zhazhazhazhi** [comment]
maybe the model ( your provided ) is train on coco .   (117V and 80 O )
HOIW ( 10 V and 11O)

**Comment by YueLiao** [comment]
> > > I  find that the MAP value is the 0.2  in the HOI-W .
> > 
> > 
> > Can you provide more details？ I can achieve 68.1 mAP based on DLA backbone using the repo.
> 
> But I find the error : in the ../models/ctdet_coco_dla_2x.pth .
> Skip loading parameter hm.2.weight, required shapetorch.Size([11, 256, 1, 1]), loaded shapetorch.Size([80, 256, 1, 1]).
> Skip loading parameter hm.2.bias, required shapetorch.Size([11]), loaded shapetorch.Size([80]).
> No param hm_rel.0.weight.
> No param hm_rel.0.bias.
> No param hm_rel.2.weight.
> No param hm_rel.2.bias.
> No param sub_offset.0.weight.
> No param sub_offset.0.bias.
> No param sub_offset.2.weight.
> No param sub_offset.2.bias.
> No param obj_offset.0.weight.
> No param obj_offset.0.bias.
> No param obj_offset.2.weight.
> No param obj_offset.2.bias.
> 
> Can you provide your pre-trained model in HOIW or HOIA dataset?

DLA model for HOI-W: https://drive.google.com/file/d/1R3Um5rsyEXDiz2ojnbmKKjpQkAbXT4Ob/view?usp=sharing
corre_npy HOI-W: https://drive.google.com/file/d/1f9nU2uyxpfPTcyDgK5mPYvqI4T84biq5/view?usp=sharing

**Comment by zhazhazhazhi** [comment]
> > > > I  find that the MAP value is the 0.2  in the HOI-W .
> > > 
> > > 
> > > Can you provide more details？ I can achieve 68.1 mAP based on DLA backbone using the repo.
> > 
> > 
> > But I find the error : in the ../models/ctdet_coco_dla_2x.pth .
> > Skip loading parameter hm.2.weight, required shapetorch.Size([11, 256, 1, 1]), loaded shapetorch.Size([80, 256, 1, 1]).
> > Skip loading parameter hm.2.bias, required shapetorch.Size([11]), loaded shapetorch.Size([80]).
> > No param hm_rel.0.weight.
> > No param hm_rel.0.bias.
> > No param hm_rel.2.weight.
> > No param hm_rel.2.bias.
> > No param sub_offset.0.weight.
> > No param sub_offset.0.bias.
> > No param sub_offset.2.weight.
> > No param sub_offset.2.bias.
> > No param obj_offset.0.weight.
> > No param obj_offset.0.bias.
> > No param obj_offset.2.weight.
> > No param obj_offset.2.bias.
> > Can you provide your pre-trained model in HOIW or HOIA dataset?
> 
> DLA model for HOI-W: https://drive.google.com/file/d/1R3Um5rsyEXDiz2ojnbmKKjpQkAbXT4Ob/view?usp=sharing
> corre_npy HOI-W: https://drive.google.com/file/d/1f9nU2uyxpfPTcyDgK5mPYvqI4T84biq5/view?usp=sharing

thank you , but the corre_npy need to be  approve (   in your email )

**Comment by YueLiao** [comment]
> > > > > I  find that the MAP value is the 0.2  in the HOI-W .
> > > > 
> > > > 
> > > > Can you provide more details？ I can achieve 68.1 mAP based on DLA backbone using the repo.
> > > 
> > > 
> > > But I find the error : in the ../models/ctdet_coco_dla_2x.pth .
> > > Skip loading parameter hm.2.weight, required shapetorch.Size([11, 256, 1, 1]), loaded shapetorch.Size([80, 256, 1, 1]).
> > > Skip loading parameter hm.2.bias, required shapetorch.Size([11]), loaded shapetorch.Size([80]).
> > > No param hm_rel.0.weight.
> > > No param hm_rel.0.bias.
> > > No param hm_rel.2.weight.
> > > No param hm_rel.2.bias.
> > > No param sub_offset.0.weight.
> > > No param sub_offset.0.bias.
> > > No param sub_offset.2.weight.
> > > No param sub_offset.2.bias.
> > > No param obj_offset.0.weight.
> > > No param obj_offset.0.bias.
> > > No param obj_offset.2.weight.
> > > No param obj_offset.2.bias.
> > > Can you provide your pre-trained model in HOIW or HOIA dataset?
> > 
> > 
> > DLA model for HOI-W: https://drive.google.com/file/d/1R3Um5rsyEXDiz2ojnbmKKjpQkAbXT4Ob/view?usp=sharing
> > corre_npy HOI-W: https://drive.google.com/file/d/1f9nU2uyxpfPTcyDgK5mPYvqI4T84biq5/view?usp=sharing
> 
> thank you , but the corre_npy need to be approve ( in your email )

I have processed it. The model is trained based on the old version repo, where the human and object branches are split. I am looking for the old version repo now.

**Comment by zhazhazhazhi** [comment]
thank you. solve it .

**Comment by tjumiao** [comment]
> > Onedrive is also good. Thanks.
> 
> https://sensetime-my.sharepoint.cn/:f:/g/personal/liaoyue_sensetime_com/EkUA1sc_sHNNmVNiVsyF8L0BCE4AXHa7sFEb8YTx11laBg?e=ArfMCD

Hi, this link shows 404 not found. Is it for the HOI-W dataset?

Thanks

**Comment by YueLiao** [comment]
> > > Onedrive is also good. Thanks.
> > 
> > 
> > https://sensetime-my.sharepoint.cn/:f:/g/personal/liaoyue_sensetime_com/EkUA1sc_sHNNmVNiVsyF8L0BCE4AXHa7sFEb8YTx11laBg?e=ArfMCD
> 
> Hi, this link shows 404 not found. Is it for the HOI-W dataset?
> 
> Thanks

If you could visit the Baidu Driver, the HOI-W could be downloaded [here](http://picdataset.com/challenge/dataset/download/). Otherwise, I will upload the HOI-W to the Google Drive today.

**Comment by tjumiao** [comment]
Hi,

Thanks for your response. It would be great if you could upload the dataset to Google Drive. The downloading speed from Baidu Driver is super slow for me.

Thank you very much!

**Comment by tjumiao** [comment]
> > > > Onedrive is also good. Thanks.
> > > 
> > > 
> > > https://sensetime-my.sharepoint.cn/:f:/g/personal/liaoyue_sensetime_com/EkUA1sc_sHNNmVNiVsyF8L0BCE4AXHa7sFEb8YTx11laBg?e=ArfMCD
> > 
> > 
> > Hi, this link shows 404 not found. Is it for the HOI-W dataset?
> > Thanks
> 
> If you could visit the Baidu Driver, the HOI-W could be downloaded [here](http://picdataset.com/challenge/dataset/download/). Otherwise, I will upload the HOI-W to the Google Drive today.

Hi,

Thanks for your response. It would be great if you could upload the dataset to Google Drive. The downloading speed from Baidu Driver is super slow for me.

Thank you very much!

**Comment by YueLiao** [comment]
> > > > > Onedrive is also good. Thanks.
> > > > 
> > > > 
> > > > https://sensetime-my.sharepoint.cn/:f:/g/personal/liaoyue_sensetime_com/EkUA1sc_sHNNmVNiVsyF8L0BCE4AXHa7sFEb8YTx11laBg?e=ArfMCD
> > > 
> > > 
> > > Hi, this link shows 404 not found. Is it for the HOI-W dataset?
> > > Thanks
> > 
> > 
> > If you could visit the Baidu Driver, the HOI-W could be downloaded [here](http://picdataset.com/challenge/dataset/download/). Otherwise, I will upload the HOI-W to the Google Drive today.
> 
> Hi,
> 
> Thanks for your response. It would be great if you could upload the dataset to Google Drive. The downloading speed from Baidu Driver is super slow for me.
> 
> Thank you very much!

https://drive.google.com/drive/folders/15xrIt-biSmE9hEJ2W6lWlUmdDmhatjKt?usp=sharing
